index.js

```diff
@@ -60,19 +60,40 @@ const Resizer = ({ array, updateCallback }) => {
   `;
 };
 
+let mouseDown = false;
+let dragType = 1;
+
 const Input = ({ array, updateCallback }) => {
-  const updateCell = (rowIndex, colIndex) => () => {
+  const updateCell = (rowIndex, colIndex) => {
     array[rowIndex][colIndex] ^= 1;
     updateCallback(array);
   };
 
+  const updateCellDrag = (rowIndex, colIndex) => {
+    array[rowIndex][colIndex] = dragType;
+    updateCallback(array);
+  };
+
   const createRow = (row, rowIndex) => html`
     <div>
       ${row.map(
         (active, colIndex) => html`
           <button
             class=${active ? "active cell" : "cell"}
-            onClick=${updateCell(rowIndex, colIndex)}
+            onClick=${() => {updateCell(rowIndex, colIndex)}}
+            onMouseDown=${() => {
+              mouseDown = true;
+              dragType = !array[rowIndex][colIndex]
+              updateCell(rowIndex, colIndex);
+            }}
+            onMouseUp=${() => {
+              mouseDown = false;
+            }}
+            onMouseOver=${() => {
+              if (mouseDown) {
```

Comment:
Instead of mouseDown, thanks to React's synthetic event handling, we can use `(e) => if (e.buttons) ...` to tell if a mouse button is being pressed during the MouseOver event.